File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/InitializationPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class InitializationPhase extends BenchmarkPhase {

    InitializationPhase(Benchmark benchmark) {
        super(benchmark);
    }

    @Override
    protected Status execute() {
        Status status = getBenchmark().initialize();

        getBenchmark().synchronize();

        return status;
    }

}

File: cpp/src/benchmark/phase/InitializationPhase.cpp
#include "InitializationPhase.h"
#include <benchmark/Benchmark.h>

#include <utility>

namespace Observatory {

InitializationPhase::InitializationPhase(Benchmark &benchmark) :
        BenchmarkPhase(benchmark) {}

const char* InitializationPhase::getName() {
    return "InitializationPhase";
}

Status InitializationPhase::execute() {
    return getBenchmark().initialize();
}

}